src/threads/scriptthread.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2014-2020 Patrizio Bekerle -- <patrizio@bekerle.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+
+#include "scriptthread.h"
+#include <QDebug>
+
+QMap<QString, int> ScriptThread::threadCounter;
+
+ScriptThread::ScriptThread(ScriptingService *ss,
```

Comment:
Can't you do that all with QFuture?

Reply:
Don't know, I'll have a look. As I warned, I'm new to Qt.

Reply:
Do you mean the entire class or just the thread counting?

Reply:
Before your pull request I thought about doing everything with `QFutureWatcher` (like in the `src/widgets/navigationwidget.cpp`). Maybe then you don't even need the whole new class...

Reply:
QFutureWatcher looks interesting, I could try rewriting it but that might currently be beyond me.

Reply:
Maybe I'll look at it later. It should only be a view lines of code (if I didn't thought wrong).

Reply:
Inheriting from `QThread` is generally not the best idea, it is usually for threads that maybe running for long periods of time. A worker thread is a better solution in many cases.

Reply:
https://doc.qt.io/qt-5/threads-technologies.html a nice overview

Reply:
I've ran across  that overview but I was missing some minimal working example.

Reply:
yes, that is missing indeed. But we have a small and simple implementation of `QtConcurrent` in `navigationWidget` so that can be used as reference.

Reply:
I'll give it a shot.

Reply:
It works fine for one thread but `QFutureWatcher` is only meant to watch a single `QFuture`. However, the function might be invoked multiple times, so the mechanism needs a vector/list that connects in addition all signal/slots accordingly.
I don't see a Qt multithreading technology for multiple single life thread with result values other than `QThread`. :confused:

Reply:
If it doesn't work then yeah `QThread` is the only choice of course.